modules/ggl-http/src/gghttp_util.c

```diff
@@ -277,6 +282,155 @@ static size_t write_response_to_fd(
     return size_of_response_data;
 }
 
+/// @brief Set HTTPS proxy configuration for curl requests if enabled or setup
+/// by the config.
+///
+/// Depending on whether HTTPS proxy is enabled, this function will attempt to
+/// add the root CA trust store to the curl's configuration. On success, it will
+/// try to optionally add the key and cert for mTLS HTTPS proxy if configured.
+///
+/// @param[in] curl_data A pointer to the curl data which is to be updated with
+/// the proxy configuration.
+/// @return GGL_ERR_OK on success, error code otherwise.
+static GglError set_curl_proxy_config(CurlData *curl_data) {
+    assert(curl_data != NULL);
+
+    if (curl_data == NULL) {
+        GGL_LOGE("Pointer to curl data cannot be NULL");
+        return GGL_ERR_FATAL;
+    }
+
+    uint8_t proxy_uri_mem[PATH_MAX] = { 0 };
+    GglArena alloc_proxy = ggl_arena_init(
+        ggl_buffer_substr(GGL_BUF(proxy_uri_mem), 0, sizeof(proxy_uri_mem) - 1)
+    );
+    GglBuffer proxy_uri;
+    GglError ret = ggl_gg_config_read_str(
+        GGL_BUF_LIST(
+            GGL_STR("services"),
+            GGL_STR("aws.greengrass.NucleusLite"),
+            GGL_STR("configuration"),
+            GGL_STR("networkProxy"),
+            GGL_STR("proxy"),
+            GGL_STR("url")
+        ),
+        &alloc_proxy,
+        &proxy_uri
+    );
+
+    if (ret == GGL_ERR_OK) {
+        proxy_uri_mem[proxy_uri.len] = '\0';
+        // Add proxy configuration
+        CURLcode curl_error
+            = curl_easy_setopt(curl_data->curl, CURLOPT_PROXY, proxy_uri_mem);
```

Comment:
Is this needed? AFAIK proxy already worked, we just need certs added for mTLS right?

Reply:
Yes, good catch. Removed.